test/runtests.jl

```diff
@@ -207,6 +207,18 @@ end
     @test_broken glm(@formula(y ~ x1 + x2), df, Normal(), IdentityLink())
 end
 
+@testset "Linear model with no intercept" begin
+    # test case to test r2 for no intercept model
+    # https://www.itl.nist.gov/div898/strd/lls/data/LINKS/DATA/NoInt1.dat
+
+    data = DataFrame(x = 60:70, y = 130:140)
+    mdl = lm(@formula(y ~ 0 + x), data)
+    @test coef(mdl) ≈ [2.07438016528926]
+    @test stderror(mdl) ≈ [0.165289256198347E-01]
+    @test GLM.dispersion(mdl.model) ≈ 3.56753034006338
+    @test r2(mdl) ≈ 0.999365492298663
```

Comment:
Can you also test other indicators (see other tests)?

Reply:
Thanks for your suggestion. While testing the `adjusted r2`, I feel we may need to modify the `adjusted r2` also. 
Let me check and test with a few more test cases.